woocommerce/woocommerce-android
# Variation details: Add image-editing functionality to variations

Fixes #2765. This PR adds the image-editing functionality to variations, which have a limit of one image. 

![image](https://user-images.githubusercontent.com/1522856/92575323-77f15480-f288-11ea-8c43-b4523189ff4b.png)

Note: Currently, the API doesn't support image removal, so the Trash icon has been removed from the image detail.

**To test:**

1. Go to Products -> Tap on a variable product -> Tap on Variations -> Select a variation without images
2. Tap on the Add image button and notice the image source dialog is displayed
3. Select an image from local storage and notice only one image at a time can be selected
4. Notice a thumbnail with a progress icon is displayed
5. Tap on Done button
6. Notice the thumbnail is also displayed in the variation detail screen
7. After the upload is finished, notice that the progress bar disappears
8. Notice that the Add image button is not visible anymore
9. Tap on the image
10. Notice the image is displayed full-screen and that there's no Remove button
11. Tap on the back button
12. Notice there is a Replace photo button at the top
13. Tap on it and use WP gallery as a source
14. Notice only one image at a time can be selected and after selection, the image is replaced immediately
15. Tap on Replace photo button again
16. Select the camera source
17. After taking a picture, notice the image is uploaded
18. Tap on Done
19. Tap on Update
20. Notice the variation is saved and the Update button is gone
